server/src/controllers/users.ts

```diff
@@ -1,7 +1,38 @@
-import type { IUser } from '@contapp/shared';
+import type { IUser, IUserForUpdate } from '@contapp/shared';
+import Services from '../services';
 import type { Reply, Request } from '../types';
+import { comparePassword, hashPassword } from '../utils/password';
 
 export async function me(request: Request, reply: Reply) {
 	const { password, ...user } = request.user as IUser;
 	return reply.code(200).send({ success: true, message: 'Ok', data: user });
 }
+
+export class UserController {
+	static async updateUser(req: Request, reply: Reply) {
+		const { id, password } = req.user as IUser;
+		const userUpdates = req.body as IUserForUpdate;
+
+		if (userUpdates.password && password) {
+			if (!(await comparePassword(userUpdates.old_password, password))) {
+				return reply
+					.status(400)
+					.send({ success: false, message: 'Invalid password', errors: [] });
```

Comment:
in the service, please remove this and use BadRequestError to create the error, automatically with thow BadRequestError(message) the response will be sent